crates/telio-wg/src/wg.rs

```diff
@@ -747,9 +823,57 @@ impl State {
         }
 
         self.interface = to.clone();
-
         success
     }
+
+    fn update_additional_peer_info(&mut self, to: &uapi::Interface) {
+        let peers: HashSet<PublicKey> = to.peers.keys().cloned().collect();
+        self.last_rx_timestamp.retain(|pk, _| peers.contains(pk));
+        self.last_link_state.retain(|pk, _| peers.contains(pk));
+
+        for peer in peers {
+            let old_rxed_bytes: Option<u64> = self.last_rx_timestamp.get(&peer).map(|ts| ts.1);
+            let new_rxed_bytes: Option<u64> = to.peers.get(&peer).and_then(|p| p.rx_bytes);
+
+            if old_rxed_bytes != new_rxed_bytes {
+                match new_rxed_bytes {
+                    Some(new_rxed_bytes) if new_rxed_bytes > 0 => {
+                        telio_log_debug!(
+                            "updating peer last rx timestamp {:?} {:?}",
+                            peer,
+                            new_rxed_bytes
+                        );
+                        self.last_rx_timestamp
+                            .insert(peer, (Instant::now(), new_rxed_bytes));
+                    }
+                    _ => {
+                        if self.last_rx_timestamp.remove(&peer).is_some() {
+                            telio_log_debug!("removed peer last rx timestamp {:?}", peer);
+                        }
+                    }
+                }
+            }
+
+            match self.last_rx_timestamp.get(&peer) {
+                Some((t, _)) => {
+                    let configurable_rtt = match self.no_link_detection {
+                        NoLinkDetection::Disabled => DEFAULT_CONFIGURABLE_RTT,
```

Comment:
Could you clarify a bit:

- `start_with()` function short-circuits when it encounters disabled link-state feature - here we provide a default value even if it's disabled and modify the state

As per my understanding we should choose to either do nothing about it if it's disabled, or provide a default value and always work.

The feature doesn't interfer with inner workings of anything else so it's pretty safe to have it always on to be fair, this way it could be even simpler.

Reply:
start_with doesn't have anything to do with this. There I just save the no_link_detection config in the State.
The short-circuit happens only when we generate events. 
As you said maintaining the last_link_state even when the feature is disabled should not infer with anything, might actually be useful I think if we will want to enable it during runtime at some point. But now that you pointed it out and I'm thinking more about it, I think we shouldn't do that due to performance reasons. This update function from telio-wg is one of the most called function in the lib. What do you think? Should I leave it as it is? Or should I skip the maintaining of last_link_state when the feature is disabled?